### PR TITLE
refactor: remove six usage

### PR DESCRIPTION
**Pull request by synchon**
* [x] description of feature/fix
* [x] tests added/passed

This PR removes `six` usage from the code base and updates installation config files.

### CODE REVIEW

**Comment by synchon** [comment]
The test is failing due to network issue when performing `test_web` tests.

File: src/cobra/test/test_core/test_dictlist.py [hunk 1]
```diff
@@ -9,7 +9,6 @@
 from pickle import HIGHEST_PROTOCOL, dumps, loads
 
 import pytest
-from six.moves import range
```
Comment (Midnighter) [inline src/cobra/test/test_core/test_dictlist.py:12]: Was `range` an unused import? It's not removed from any statement in the module.

File: src/cobra/test/test_flux_analysis/test_variability.py [hunk 2]
```diff
@@ -6,7 +6,6 @@
 
 import numpy as np
 import pytest
-from six import iteritems
```
Comment (Midnighter) [inline src/cobra/test/test_flux_analysis/test_variability.py:9]: Again an unused import?

**Review by Midnighter — APPROVED** [review]
Looks good! It's time we enabled flake8 so that unused imports and variables would be caught immediately.

File: src/cobra/test/test_core/test_dictlist.py (hunk 1 above)
Comment (synchon) [reply src/cobra/test/test_core/test_dictlist.py:12]: I think it shadowed Python 3's `range`.

File: src/cobra/test/test_flux_analysis/test_variability.py (hunk 2 above)
Comment (synchon) [reply src/cobra/test/test_flux_analysis/test_variability.py:9]: Yes this one was unused.

**Comment by synchon** [comment]
I would say we wait till we have completely fixed all the existing `flake8` issues. This would help to catch only the future issues and not the old ones.

**Pull request merged**